test/modules/Modules.Test/TestResultCoordinator/Reports/DirectMethod/DirectMethodReportGeneratorTest.cs

```diff
@@ -0,0 +1,507 @@
+                new object[]
+                {
+                    Enumerable.Range(1, 7).Select(v => v.ToString()),
+                    new[] { "1", "2", "3", "4", "5", "6", "7" },
+                    new int[] { 200, 200, 200, 200, 200, 200, 200 },
+                    new DateTime[]
+                    {
+                        new DateTime(2020, 1, 1, 9, 10, 12, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 13, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 16, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 22, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 23, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
+                    },
+                    10, 6, 0, 0, 0, 0, 1, 0, 0, false
+                },
+                new object[]
+                {
+                    Enumerable.Range(1, 7).Select(v => v.ToString()),
+                    new[] { "1", "2", "3", "5", "6", "7" },
+                    new int[] { 200, 200, 200, 200, 200, 200, 200 },
+                    new DateTime[]
+                    {
+                        new DateTime(2020, 1, 1, 9, 10, 12, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 13, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 21, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 22, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 23, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
+                    },
+                    10, 6, 0, 0, 0, 0, 0, 1, 0, true
+                },
+                new object[]
+                {
+                    Enumerable.Range(1, 6).Select(v => v.ToString()),
+                    new[] { "1", "2", "3", "4", "5", "6", "7" },
+                    new int[] { 200, 200, 200, 200, 200, 200, 200 },
+                    new DateTime[]
+                    {
+                        new DateTime(2020, 1, 1, 9, 10, 12, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 13, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 21, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 22, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 23, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 10),
+                        new DateTime(2020, 1, 1, 9, 10, 24, 15)
+                    },
+                    10, 6, 0, 0, 0, 0, 0, 0, 1, false
+                }
+            };
+
+        static Option<NetworkStatusTimeline> NetworkStatusTimeline => MockNetworkStatusTimeline.GetMock(new TimeSpan(0, 0, 0, 0, 5)).Result;
+
+        [Fact]
+        public void TestConstructorSuccess()
+        {
+            string expectedSource = "expectedSource";
+            string actualSource = "actualSource";
+            int batchSize = 10;
```

Comment:
In my opinion, unit tests should be exactly reproducible. If we wanted to add what you are talking about, that is more like fuzz testing. It could definitely have its place, but I don't think in this PR. But I am also interested to hear what others think

Reply:
Unit tests should be exactly reproduced.  Sometimes, you may see something like Args.Any<int> in some unit tests for some arguments which are not important for that unit test.  We can discuss fuzz testing in team meeting or retrospective meeting.

Reply:
Gotcha.